enterprise/server/backends/migration_cache/migration_cache.go

```diff
@@ -714,7 +727,19 @@ func (mc *MigrationCache) logCopyChanFullInBackground() {
 func (mc *MigrationCache) copy(c *copyData) {
 	defer c.ctxCancel()
 
-	srcReader, err := mc.src.Reader(c.ctx, c.d, 0, 0)
+	cache, err := mc.WithIsolation(c.ctx, c.cacheType, c.remoteInstanceName)
```

Comment:
You can add a (private) `withIsolation` method that returns a *MigrationCache instead of interfaces.Cache and call that instead to avoid needing to cast. `WithIsolation` can then also call the private `withIsolation` method.

Reply:
Done